multi3.c

```diff
@@ -0,0 +1,206 @@
+#if DEBUG
+#include <assert.h>
+#endif
+#include <math.h>
+#include <pthread.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include "utils.h"
+
+// Global because they're heavily shared and don't change throughout run, so its
+// a good idea
+
+
+int p, d, maxfe, n, b, k;
+double (*f)(double x[]);
+
+typedef struct {
+  int start;
+  int end;
+  int iters;
+} params;
+
+double *solutions;
+double *all_fits;
+
+// Shared vars, lock plz
+double *global_best_sol, *global_worst_sol;
+double global_min_fit = INFINITY;
+double global_max_fit = 0;
+
+void *jaya_sub(void *point) {
+  params *par = (params *)point;
+  printf("start:%i end:%i iter:%i\n", par->start, par->end, par->iters);
+
+  double *r1 = malloc(sizeof(double) * d);
+  double *r2 = malloc(sizeof(double) * d);
+  double *xi = malloc(sizeof(double) * d);
+  double *best_sol = malloc(sizeof(double) * d);
+  double *worst_sol = malloc(sizeof(double) * d);
+
+  double min_fit = INFINITY, max_fit = 0;
+  double fit, x, new_fit, old_fit;
+  int min_idx = -1, max_idx = -1;
+  int iter, i, j;
+
+  // determining best and worst sol at the start
+  for (i = par->start; i < par->end; i++) {
```

Comment:
Aren't we skipping the very last element like this? I could be wrong

Reply:
I am confused on the math for this, because I believe it will depend on divisibility of the iter_per_merge on the thread count